### ISSUE TITLE
Setting parse to true gives error while using webhooks

### ISSUE THREAD

**Issue by MD-AZMAL** [ory/kratos#3177]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Network](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

I am using kratos v0.11.0. I am trying to use webhooks in the registration flow, and abort the flow when the webhook gives error saying parse is not allowed.

### Reproducing the bug

1. Use the above kratos.yaml to run kratos
2. kratos wont be able to start and will give the above error

### Relevant log output

```shell
The configuration contains values or keys which are invalid:
selfservice.flows.registration.after.password.hooks.0: map[config:map[body:base64://ZnVuY3Rpb24oY3R4KSB7IAogICAgY3R4OiBjdHgKIH0KCgo= method:POST response:map[parse:false] url:http://host.docker.internal:4000/api/v1/web-hooks/ory-identities/check-user-in-invite] hook:web_hook]
                                                       ^-- additionalProperties "config" not allowed

The configuration contains values or keys which are invalid:
selfservice.flows.registration.after.password.hooks.0: map[config:map[body:base64://ZnVuY3Rpb24oY3R4KSB7IAogICAgY3R4OiBjdHgKIH0KCgo= method:POST response:map[parse:false] url:http://host.docker.internal:4000/api/v1/web-hooks/ory-identities/check-user-in-invite] hook:web_hook]
                                                       ^-- doesn't validate with "#/definitions/selfServiceWebHook"

The configuration contains values or keys which are invalid:
selfservice.flows.registration.after.password.hooks.0.config.response: map[parse:false]
                                                                       ^-- additionalProperties "parse" not allowed

time=2023-03-19T12:12:58Z level=error msg=Unable to instantiate configuration. audience=application error=map[message:I[#/selfservice/flows/registration/after] S[#/properties/selfservice/properties/flows/properties/registration/properties/after/$ref] doesn't validate with "#/definitions/selfServiceAfterRegistration"
  I[#/selfservice/flows/registration/after/password] S[#/definitions/selfServiceAfterRegistration/properties/password/$ref] doesn't validate with "#/definitions/selfServiceAfterRegistrationMethod"
    I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceAfterRegistrationMethod/properties/hooks/items/anyOf] anyOf failed
      I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceAfterRegistrationMethod/properties/hooks/items/anyOf/0/$ref] doesn't validate with "#/definitions/selfServiceSessionIssuerHook"
        I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceSessionIssuerHook] validation failed
          I[#/selfservice/flows/registration/after/password/hooks/0/hook] S[#/definitions/selfServiceSessionIssuerHook/properties/hook/const] value must be "session"
          I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceSessionIssuerHook/additionalProperties] additionalProperties "config" not allowed
      I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceAfterRegistrationMethod/properties/hooks/items/anyOf/1/$ref] doesn't validate with "#/definitions/selfServiceWebHook"
        I[#/selfservice/flows/registration/after/password/hooks/0/config/response] S[#/definitions/selfServiceWebHook/properties/config/properties/response/additionalProperties] additionalProperties "parse" not allowed] service_name=Ory Kratos service_version=v0.11.0
Error: I[#/selfservice/flows/registration/after] S[#/properties/selfservice/properties/flows/properties/registration/properties/after/$ref] doesn't validate with "#/definitions/selfServiceAfterRegistration"
  I[#/selfservice/flows/registration/after/password] S[#/definitions/selfServiceAfterRegistration/properties/password/$ref] doesn't validate with "#/definitions/selfServiceAfterRegistrationMethod"
    I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceAfterRegistrationMethod/properties/hooks/items/anyOf] anyOf failed
      I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceAfterRegistrationMethod/properties/hooks/items/anyOf/0/$ref] doesn't validate with "#/definitions/selfServiceSessionIssuerHook"
        I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceSessionIssuerHook] validation failed
          I[#/selfservice/flows/registration/after/password/hooks/0/hook] S[#/definitions/selfServiceSessionIssuerHook/properties/hook/const] value must be "session"
          I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceSessionIssuerHook/additionalProperties] additionalProperties "config" not allowed
      I[#/selfservice/flows/registration/after/password/hooks/0] S[#/definitions/selfServiceAfterRegistrationMethod/properties/hooks/items/anyOf/1/$ref] doesn't validate with "#/definitions/selfServiceWebHook"
        I[#/selfservice/flows/registration/after/password/hooks/0/config/response] S[#/definitions/selfServiceWebHook/properties/config/properties/response/additionalProperties] additionalProperties "parse" not allowed
```

### Relevant configuration

```yml
version: v0.7.1-alpha.1


serve:
  public:
    base_url: http://127.0.0.1:3433/
    cors:
      enabled: true
      allowed_origins:
        - http://127.0.0.1:4000
        - http://localhost:4000
  admin:
    base_url: http://kratos-oiconomos:4434/

selfservice:
  default_browser_return_url: http://127.0.0.1:3000/
  allowed_return_urls:
    - http://127.0.0.1:3000

  methods:
    password:
      enabled: true

  flows:
    error:
      ui_url: http://127.0.0.1:3000/error

    settings:
      ui_url: http://127.0.0.1:3000/settings
      privileged_session_max_age: 15m

    recovery:
      enabled: true
      ui_url: http://127.0.0.1:3000/recovery

    verification:
      enabled: true
      ui_url: http://127.0.0.1:3000/verification
      after:
        default_browser_return_url: http://127.0.0.1:3000/

    logout:
      after:
        default_browser_return_url: http://127.0.0.1:3000/login

    login:
      ui_url: http://127.0.0.1:3000/login
      lifespan: 10m

    registration:
      lifespan: 10m
      ui_url: http://127.0.0.1:3000/registration
      after:
        password:
          hooks:
            - hook: web_hook
              config:
                url: http://host.docker.internal:4000/api/v1/web-hooks/ory-identities/check-user-in-invite
                method: POST
                body: base64://ZnVuY3Rpb24oY3R4KSB7IAogICAgY3R4OiBjdHgKIH0KCgo=
                response:
                  parse: false
            - hook: session

log:
  level: debug
  format: text
  leak_sensitive_values: true

secrets:
  cookie:
    - PLEASE-CHANGE-ME-I-AM-VERY-INSECURE
  cipher:
    - 32-LONG-SECRET-NOT-SECURE-AT-ALL

ciphers:
  algorithm: xchacha20-poly1305

hashers:
  algorithm: bcrypt
  bcrypt:
    cost: 8

identity:
  default_schema_id: default
  schemas:
    - id: default
      url: file:///etc/config/kratos/identity.schema.json

courier:
  smtp:
    connection_uri: smtps://test:test@mailslurper:1025/?skip_ssl_verify=true
```

### Version

0.11.0

### On which operating system are you observing this issue?

Windows

### In which environment are you deploying?

Docker Compose

**Comment by ChrisLahaye** [comment]
This feature isn't part of any release yet. You will have to build Kratos from the master branch yourself.

**Comment by kmherrmann** [comment]
This should be resolved in v0.13.0
